File: src/main/java/com/developerphil/adbidea/ui/MyApplistModel.java
package com.developerphil.adbidea.ui;

import java.util.List;
import javax.swing.AbstractListModel;

/**
 * Created by XQ Yang on 10/8/2018  5:41 PM.
 * Description :
 */

class MyApplistModel extends AbstractListModel<String> {

    private List<String> mList;

    public MyApplistModel(List<String> list) {
        mList = list;
    }

    @Override
    public int getSize() {
        return mList.size();
    }

    @Override
    public String getElementAt(int index) {
        return mList.get(index);
    }

    public void delete(String s) {
        int index = mList.indexOf(s);
        if (index != -1) {
            mList.remove(index);
            fireIntervalRemoved(this, index, index);
        }
    }
}

File: src/main/kotlin/com/developerphil/adbidea/ui/MyApplistModel.kt
package com.developerphil.adbidea.ui

import javax.swing.AbstractListModel

/**
 * Created by XQ Yang on 10/8/2018  5:41 PM.
 * Description :
 */

internal class MyApplistModel(private val mList: MutableList<String>) : AbstractListModel<String>() {

    override fun getSize(): Int {
        return mList.size
    }

    override fun getElementAt(index: Int): String {
        return mList[index]
    }

    fun delete(s: String) {
        val index = mList.indexOf(s)
        if (index != -1) {
            mList.removeAt(index)
            fireIntervalRemoved(this, index, index)
        }
    }
}
